subsys/bluetooth/mesh/lpn.c

```diff
@@ -299,20 +313,32 @@ static void friend_req_sent(uint16_t duration, int err, void *user_data)
 		return;
 	}
 
-	lpn->adv_duration = duration;
+	lpn->adv_duration = k_uptime_get_32() - lpn->adv_duration;
 
 	if (IS_ENABLED(CONFIG_BT_MESH_LPN_ESTABLISHMENT)) {
-		k_work_reschedule(&lpn->timer, K_MSEC(FRIEND_REQ_WAIT));
+		k_work_reschedule(&lpn->timer,
+				  K_MSEC(FRIEND_REQ_WAIT - (int32_t)lpn->adv_duration));
 		lpn_set_state(BT_MESH_LPN_REQ_WAIT);
 	} else {
-		k_work_reschedule(&lpn->timer,
-				  K_MSEC(duration + FRIEND_REQ_TIMEOUT));
+		k_work_reschedule(&lpn->timer, K_MSEC(FRIEND_REQ_TIMEOUT));
 		lpn_set_state(BT_MESH_LPN_WAIT_OFFER);
 	}
 }
 
-static const struct bt_mesh_send_cb friend_req_sent_cb = {
-	.start = friend_req_sent,
+static void friend_req_send_start(uint16_t duration, int err, void *user_data)
+{
+	struct bt_mesh_lpn *lpn = &bt_mesh.lpn;
+
+	lpn->adv_duration = k_uptime_get_32();
```

Comment:
Using a variable called "duration" for something that's not a duration looks a bit unintuitive (in this case it's just a timestamp, until you calculate an actual duration by taking the diff of two timestamps in the send_end callback).

Maybe consider renaming the variable to something more generic, e.g. `adv_time`, if you want to keep using it for these two different purposes?